docs/nextflow/reduce-api-calls.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Reducing Wave API calls
+description: Learn how to freeze containers and reduce API calls in large-scale Nextflow pipelines
+date created: 2025-09-30
+date edited: 2025-10-31
+tags: [nextflow, wave, rate limits, guides]
+---
+
+Large-scale pipelines that pull container images across thousands of concurrent tasks can encounter Wave rate limits. This guide describes how to use Wave freeze to reduce API calls and avoid rate limits.
+
+## API limits
+
+Wave applies rate limits to container builds and pulls (manifest requests). Authenticated users have higher rate limits than anonymous users.
+
+If you provide a Platform access token, the following rate limits apply:
+
+- 250 container builds per hour
+- 2,000 container pulls (manifest requests) per minute
+
+If you don't provide a Platform access token, the following rate limits apply:
+
+- 25 container builds per day
+- 100 container pulls (manifest requests) per hour
+
+## How Wave pull rate limits work
+
+When you pull a container image:
+
+1. The Docker service downloads the container manifest (a JSON file detailing each layer of the final image).
+1. The Docker service downloads each layer listed in the manifest file.
+
+Wave defines a pull as downloading both the container manifest and all layers. Therefore:
+
+- The manifest request to Wave counts as one pull against your rate limit
+- Layer and blob requests don't count against rate limits
+- A container image with 100 layers counts as 1 pull
+
+Rate limits affect pipelines with high concurrency. The following example demonstrates this issue:
+
+- 50 concurrent pipeline runs
+- Each run spawns 10,000 tasks
+- Each task pulls a container image
+- 500,000 manifest requests are made
+
+This volume exceeds the 2,000 container pulls per minute limit and causes failed tasks and pipeline errors.
+
+## How Wave freeze avoids rate limits
+
+Wave freeze builds your container image once and stores it in your registry. After the initial build, the source of the container image manifest and layers are redirected to your private registry by Wave.
+
+### Building without Wave freeze
+
+When you run your pipeline without Wave freeze:
+
+1. Each task requests a manifest from Wave.
+1. Wave performs one of the following actions:
+    - Retrieves the base image manifest from the source registry
+    - Builds the image from a Dockerfile
+    - Builds the image from a Conda definition
+1. Wave injects the Fusion layer to the container image manifest.
+1. Wave stores the final manifest on Seqera infrastructure.
+1. Wave returns the modified manifest.
+1. Every task creates one API call to Wave.
```

Comment:
I would lose this bullet -- you have a good description of how we charge API calls above, and this now mixes build mechanics with how we rate-limit.